qdrant/v1.2.x/filtering.md

```diff
@@ -516,6 +516,186 @@ client.scroll(
+                }
+            },
+            {
+                "key": "data[].b",
+                  "match": {
+                    "value": 2
+                }
+            }
+        ]
+    }
+}
+```
+
+```python
+client.scroll(
+    collection_name="{collection_name}",
+    scroll_filter=models.Filter(
+        must=[
+            models.FieldCondition(
+                key="data[].a",
+                match=models.MatchValue(value=1")
+            ),
+            models.FieldCondition(
+                key="data[].b",
+                match=models.MatchValue(value=2)
+            ),
+        ],
+    ),
+)
+```
+
+This happens because both points are matching the conditions.
+
+To retrieve only the point with id 1, you would need to use a nested object filter.
+
+Nested object filters allow arrays of objects to be queried independently of each other.
+
+It is achieved by adding a new filter condition type called `nested` formed by a payload path to focus on and a filter to apply.
+
+```http
+POST /collections/{collection_name}/points/scroll
+
+{
+    "filter": {
+        "must": [
+            "nested": {
+                {
+                    "key": "data",
+                    "filter":{
+                        "must": [
+                            {
+                                "key": "a",
+                                "match": {
+                                    "value": 1
+                                }
+                            },
+                            {
+                                "key": "b",
+                                "match": {
+                                    "value": 2
+                                }
+                            }
+                        ]
+                    }
+                }
+            }
+        ]
+    }
+}
+```
+
+```python
+client.scroll(
+    collection_name="{collection_name}",
+    scroll_filter=models.Filter(
+        must=[
+            models.NestedContainer(
+                nested=models.NestedCondition(
+                    key="data",
+                    filter=(
+                        must=[
+                            models.FieldCondition(
+                                key="a",
+                                match=models.MatchValue(value=1")
+                            ),
+                            models.FieldCondition(
+                                key="b",
+                                match=models.MatchValue(value=2)
+                            ),
+                        ]
+                    )
+                )
+            )
+        ],
+    ),
+)
+```
+
+The matching logic is modified to be applied at the level of an array element within the payload.
+
+It can be used with `must`, `must_not` and `should` with the following semantic:
+
+- nested `must`: at least a single element matches all conditions.
+- nested `must_not`: at least a single element does not match any of the conditions.
```

Comment:
The semantic of `must_not` for nested is a bit twisted, I am not sure I have a better description.
Check out this example https://github.com/qdrant/qdrant/blob/master/openapi/tests/openapi_integration/test_nested_payload_query.py#L303

Reply:
I feel like the my confusion comes from the word `any`, rather than the phrasing, how about including the equivalence
```
On a per element basis: 
- nested `must_not`: at least a single element does not match all of the conditions. 
     Can be seen as `'(A * B * C)` or `'A + 'B + 'C` 
```
or is it `'A * 'B * 'C` ?